src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1338,6 +1297,39 @@ internal static string GetModulePath()
             return currentModulePath;
         }
 
+#if !UNIX
+        /// <summary>
+        /// Returns a PSModulePath suiteable for Windows PowerShell by removing this PowerShell's specific
+        /// paths from current PSModulePath.
+        /// </summary>
+        /// <returns>
+        /// Returns appropriate PSModulePath for Windows PowerShell.
+        /// </returns>
+        internal static string GetWindowsPowerShellModulePath()
+        {
+            string currentModulePath = GetModulePath();
+
+            if (currentModulePath != null)
+            {
+                currentModulePath = currentModulePath.Replace(GetPersonalModulePath(), string.Empty).Replace(GetSharedModulePath(), string.Empty).Replace(GetPSHomeModulePath(), string.Empty);
```

Comment:
This doesn't consider the module paths defined in config.json file.
In `SetModulePath`, you can find that the `allUsersModulePath` and `personalModulePath` are retrieved from the configuration file. Should those paths be passed to powershell.exe?

Reply:
Those paths should also be scrubbed before starting powershell.exe, will update.